openstack-scripts/create-cluster.sh

```diff
@@ -251,6 +251,8 @@ function main() {
     if [[ $DOWN == "true" ]]; then
         exit 0
     fi
+    # if anything breaks from this point onwards, make sure to keep the env clean
+    trap delete-previous-cluster EXIT
```

Comment:
I don't think these PRs should be linked.

delete-previous-cluster trap doesn't have to be necessarily set here. It can just as well be at the beginning of main and thus you don't have to base your work on the previous PR. 

For one: delete-previous-cluster checks the WINDOWS_NODES, LINUX_NODES, ANSIBLE_MASTER variables. If they are not set ( i.e: script crashes before reading config file ), it does nothing. If they are set, delete-instance actually checks that the instance exists before deleting it, thus we won't have problems if trying to delete unexistent instances.

Reply:
@papagalu